docs/multicast-guide.md

```diff
@@ -0,0 +1,154 @@
+and [ACNP for multicast egress traffic](antrea-network-policy.md#acnp-for-multicast-egress-traffic)
+examples in the Antrea NetworkPolicy document.
+
+## Debugging and collecting multicast statistics
+
+Antrea provides tooling to check multicast group information and multicast traffic statistics.
+
+### Pod multicast group information
+
+The `kubectl get multicastgroups` command prints multicast groups joined by Pods in the cluster.
+Example output of the command:
+
+```bash
+kubectl get multicastgroups
+GROUP       PODS
+225.1.2.3   default/mcjoin, namespace/pod
+224.5.6.4   default/mcjoin
+```
+
+### Inbound and outbound multicast traffic statistics
+
+`antctl` supports printing multicast traffic statistics of Pods. Please refer to the corresponding [antctl user guide section](antctl.md#multicast-commands).
+
+### Multicast NetworkPolicy statistics
+
+[The Antrea NetworkPolicyStats feature](feature-gates.md#networkpolicystats) also applies to multicast NetworkPolices.
+
+## Use case example
+
+[VLC](https://www.videolan.org/vlc/) is an open source multimedia player, it can be used for video streaming.
+
+In this section, we will show how multicast video steaming works in
+Antrea using VLC.
+
+To start a video streaming server, we start a VLC Pod to stream a sample video
+to the multicast IP address `239.255.12.42` with TTL 6.
+
+```bash
+kubectl run -i --tty --image=quay.io/galexrt/vlc:latest vlc-sender -- vlc --intf ncurses --vout dummy --aout dummy sample-video.mp4 --sout udp:239.255.12.42 --ttl 6 --repeat
+```
+
+You can verify multicast traffic is sent out from this Pod
+by running `antctl get podmulticaststats` in the `antrea-agent` Pod on the local Node,
+which indicates the VLC Pod is sending video streams out.
+
+You can also check the multicast routes on the Node by running command `ip mroute`,
+which should print the following route for forwarding the multicast traffic from
+the Antrea gateway interface to the transport interface.
+
+```bash
+ip mroute
+(pod-ip, 239.255.12.42)     Iif: antrea-gw0 Oifs: transport interfaces State: resolved
+```
+
+We also create a VLC Pod to be the receiver with the following command:
+
+```bash
+kubectl run -i --tty --image=quay.io/galexrt/vlc:latest vlc-receiver -- vlc --intf ncurses --vout dummy --aout dummy udp://@239.255.12.42 --repeat
+```
+
+It's expected to see inbound multicast traffic to this Pod by running
+`antctl get podmulticaststats` in the local `antrea-agent` Pod,
+which indicates the VLC Pod is receiving the video stream.
+
+Also, the `kubectl get multicastgroups` command will show that `vlc-receiver`
+has joined multicast group `239.255.12.42`.
+
+## Limitations
+
+This feature is currently only supported for IPv4 Linux clusters.
+Support for Windows and IPv6 will be added in the future.
+
+### Encap mode
+
+Multicast traffic from Pods to external hosts is not supported in encap mode.
```

Comment:
We should support Pod-to-external multicast traffic in encap mode. The only limitation is we do not support multiple multicast interfaces on the Node, and the packet is SNATed with Node's IP after it leaves the Node with encap mode.

Reply:
@wenyingd : please provide a correct description in your mind.

Reply:
I would like to use this description with encap mode,

```
The Pod-to-external multicast traffic leaves Node from the network interface  which the default route is configured on, and the packet is SNATed. Configuration `multicastInterfaces` is not supported with encap mode.
```

Reply:
How about just
```
Configuration `multicastInterfaces` is not supported with encap mode.
```
The SNAT doesn't seem to be a limitation.